EncounterMessageManager.swift

```diff
@@ -1,194 +1,139 @@
     let userDefaultsAdvtKey = "ADVT_DATA"
     let userDefaultsAdvtExpiryKey = "ADVT_EXPIRY"
-
+    
     static let shared = EncounterMessageManager()
-
-    lazy var functions = Functions.functions(region: "asia-east2")
-
+    
     var tempId: String? {
-        guard var tempIds = UserDefaults.standard.array(forKey: userDefaultsTempIdArrayKey) as! [[String: Any]]? else {
-            return "not_found"
-        }
-
-        if let bmExpiry = tempIds.first?["expiryTime"] as? Date {
-            while Date() > bmExpiry {
-                tempIds.removeFirst()
-            }
-        }
-
-        guard let validBm = tempIds.first?["tempID"] as? String else { return "" }
-        return validBm
+        return UserDefaults.standard.string(forKey: userDefaultsTempIdKey)
     }
-
-    var advtPayload: Data? {
+    
+    var advertisedPayload: Data? {
         return UserDefaults.standard.data(forKey: userDefaultsAdvtKey)
     }
-
+    
     // This variable stores the expiry date of the broadcast message. At the same time, we will use this expiry date as the expiry date for the encryted advertisement payload
-    var advtPayloadExpiry: Date? {
+    var advertisementPayloadExpiry: Date? {
         return UserDefaults.standard.object(forKey: userDefaultsAdvtExpiryKey) as? Date
     }
-
+    
     func setup() {
         // Check payload validity
-        if advtPayloadExpiry == nil ||  Date() > advtPayloadExpiry! {
-
-            fetchBatchTempIdsFromFirebase { [unowned self](error: Error?, resp: (tempIds: [[String: Any]], refreshDate: Date)?) in
+        if advertisementPayloadExpiry == nil ||  Date() > advertisementPayloadExpiry! {
+            // Call API to get new tempId and expiry date
+            fetchTempIdFromApi { [unowned self] (error: Error?, resp:(tempId: String, expiry: Date)?) in
                 guard let response = resp else {
-                    Logger.DLog("No response, Error: \(String(describing: error))")
+                    DLog("No response, Error: \(String(describing: error))")
                     return
                 }
-                _ = self.setAdvtPayloadIntoUserDefaultsv2(response)
-                UserDefaults.standard.set(response.tempIds, forKey: self.userDefaultsTempIdArrayKey)
-
+                _ = self.setAdvertisementPayloadIntoUserDefaults(response)
+                UserDefaults.standard.set(response.tempId, forKey: self.userDefaultsTempIdKey)
             }
-
         }
     }
-
+    
     func getTempId(onComplete: @escaping (String?) -> Void) {
-        // Check refreshDate
-        if advtPayloadExpiry == nil ||  Date() > advtPayloadExpiry! {
-            fetchBatchTempIdsFromFirebase { [unowned self](error: Error?, resp: (tempIds: [[String: Any]], refreshDate: Date)?) in
+        // check refreshDate
+        if advertisementPayloadExpiry == nil ||  Date() > advertisementPayloadExpiry! {
+            fetchTempIdFromApi { [unowned self] (error: Error?, resp:(tempId: String, expiry: Date)?) in
```

Comment:
The [BlueTrace whitepaper](https://bluetrace.io/static/bluetrace_whitepaper-938063656596c104632def383eb33b3c.pdf) recommends:

>In order to ensure that devices have a supply of valid TempIDs even when the internet connection is unstable, devices pull batches of forward-dated TempIDs from the health authority’s back-end service each time.

Cut for deadlines?

Reply:
Android does the same.
Api call: https://github.com/AU-COVIDSafe/mobile-android/blob/b827cf3ccef72a3d38c6fc37466a99868823540f/app/src/main/java/au/gov/health/covidsafe/interactor/usecase/UpdateBroadcastMessageAndPerformScanWithExponentialBackOff.kt#L47
And if tempID is expired, don't broadcast: https://github.com/AU-COVIDSafe/mobile-android/blob/b827cf3ccef72a3d38c6fc37466a99868823540f/app/src/main/java/au/gov/health/covidsafe/streetpass/StreetPassWorker.kt#L586